File: dextraChallenge/Model/Repository.m
//
//  NSObject+Repository.m
//  dextraChallenge
//
//  Created by Joao Pedro Fabiano Franco on 10/11/16.
//  Copyright © 2016 joaopedroffranco. All rights reserved.
//

#import "Repository.h"
#import "dextraChallenge-Swift.h"

@implementation Repository: NSObject

-(id) initWithDictionary:(NSDictionary *)dictionary {
    self = [super init];
    self.idRepo = [dictionary objectForKey:@"id"];
    self.name = [dictionary objectForKey:@"name"];
    self.fullname = [dictionary objectForKey:@"full_name"];
    self.info = [dictionary objectForKey:@"description"];
    self.url = [dictionary objectForKey:@"html_url"];
    self.language = [dictionary objectForKey:@"language"];
    self.contributorsUrl = [dictionary objectForKey:@"contributors_url"];
    self.forks = [dictionary objectForKey:@"forks_count"];
    self.startgazers = [dictionary objectForKey:@"stargazers_count"];
    self.owner = [[User alloc] initWithDictionary:[dictionary objectForKey:@"owner"]];
    
    
    if (self.language == (NSString *)[NSNull null]){
        self.language = nil;
    }
    
    if (self.info == (NSString *)[NSNull null]){
        self.info = nil;
    }
    
    return self;
}


@end

File: dextraChallenge/Model/Repository.swift
//
//  Repository.swift
//  dextraChallenge
//
//  Created by Joao Pedro Fabiano Franco on 09/11/16.
//  Copyright © 2016 joaopedroffranco. All rights reserved.
//

import Foundation

class Repository: NSObject {
    
    var id: NSNumber!
    var name: String!
    var fullName: String?
    var info: String?
    var createdAt: Date!
    var language: String?
    var url: String!
    var score: NSNumber!
    var isPrivate: Bool!
    
    var owner: User!
    
    init(dictionary: [String : Any]){
        self.id = dictionary["id"] as! NSNumber!
        self.name = dictionary["name"] as! String
        self.fullName = dictionary["full_name"] as? String
        self.info = dictionary["description"] as? String
        self.createdAt = Date.date((dictionary["created_at"] as! String))
        self.language = dictionary["language"] as? String
        self.url = dictionary["url"] as! String
        self.score = dictionary["score"] as! NSNumber!
        self.isPrivate = dictionary["private"] as! Bool
        
        self.owner = User(dictionary: dictionary["owner"] as! [String : Any])
    }
    
    class func object(repositoriesDictionary: [[String : Any]]) -> [Repository]{
        var repositories: [Repository] = []
        for repositoryDictionary in repositoriesDictionary {
            repositories.append(Repository(dictionary: repositoryDictionary))
        }
        
        return repositories
    }
    
}
